src/popover/popover.ts

```diff
@@ -96,6 +103,33 @@ export class NgbPopover implements OnInit, OnDestroy {
     });
   }
 
+  @HostListener('document:click', ['$event'])
+  /**
+   * Determines if a click occurs outside of the popover.
+   */
+  private _docClickListener(e) {
+    if (this._triggered) {
+      return this._triggered = false;
+    }
+
+    if (!this._windowRef) {
+      return;
+    }
+
+    if (e.target !== this._windowRef.location.nativeElement &&
+        !this._windowRef.location.nativeElement.contains(e.target)) {
+      let ocEvent = document.createEvent('Event');
+      if (ocEvent.initEvent) {
+        /* For IE */
+        ocEvent.initEvent('outsideClick', true, true);
+      } else {
+        /* Avoid deprecated API */
+        ocEvent = new Event('outsideClick');
+      }
+      this._elementRef.nativeElement.dispatchEvent(ocEvent);
```

Comment:
It's dispatching an event with the name 'clickOutside' so that the listeners already set up to check for events passed to the `triggers` attribute handle the close action.

If I were to call `close()` on the popover from here, I'd have to use the trigger parsing utility to check if the close trigger name is equal to 'clickOutside.' Since the listener logic is already doing this, I figured dispatching an event would be the more logical solution.